p2p/p2p.go

```diff
@@ -198,50 +327,28 @@ func (s *Service) NoBootstrapping() bool {
 	return s.host.noBootstrap
 }
 
-// parseMessage reads message length, message type, decodes message based on type, and returns the decoded message
-func parseMessage(stream net.Stream) (Message, []byte, error) {
-	defer func() {
-		if err := stream.Close(); err != nil {
-			log.Error("fail to close stream", "error", err)
-		}
-	}()
-
-	log.Debug("got stream", "peer", stream.Conn().RemotePeer())
+// parseMessage reads message from the provided stream
+func parseMessage(stream net.Stream) (Message, error) {
 
 	rw := bufio.NewReadWriter(bufio.NewReader(stream), bufio.NewWriter(stream))
-	lengthByte, err := rw.Reader.ReadByte()
-	if err != nil {
-		log.Error("failed to read message length", "peer", stream.Conn().RemotePeer(), "error", err)
-		return nil, nil, err
-	}
 
-	// decode message length using LEB128
-	length := LEB128ToUint64([]byte{lengthByte})
-
-	// read message type byte
-	msgType, err := rw.Reader.Peek(1)
+	// check read byte
+	_, err := rw.Reader.ReadByte()
 	if err != nil {
-		log.Error("failed to read message type", "err", err)
-		return nil, nil, err
+		return nil, err
 	}
 
-	// read entire message
-	rawMsg, err := rw.Reader.Peek(int(length))
+	// check message type
+	_, err = rw.Reader.Peek(1)
 	if err != nil {
-		log.Error("failed to read message", "err", err)
-		return nil, nil, err
+		return nil, err
```

Comment:
I don't think this is needed any more. `Peek` shouldn't change anything, iirc this was to check what type of message it was, which is no longer necessary.

Reply:
Removed.